daemons/controld/controld_control.c

```diff
@@ -79,11 +79,10 @@ do_ha_control(long long action,
         crm_set_status_callback(&peer_update_callback);
         crm_set_autoreap(FALSE);
 
-        if (is_corosync_cluster()) {
 #if SUPPORT_COROSYNC
-            registered = crm_connect_corosync(cluster);
+        registered = crm_connect_corosync(cluster);
 #endif
-        }
+
```

Comment:
The is_corosync_cluster() design was kept to preserve the ability to add support for another stack in the future. It would be possible to compile in support for multiple stacks, so the currently used one would need to be detected.

Reply:
Ah, I missed that. In that case, I think the instance in crm_connect_corosync() is the one that should go. In controld_control.c, the idea would be: if is_STACK1 connect_STACK1 else if is_STACK2 connect_STACK2 etc., so registered would only be set once. While the instance in crm_connect_corosync() could be kept as a failsafe, the name clearly indicates the function should only be called for a corosync cluster, so I think we could drop it.